playbulb-candle/app.js

```diff
@@ -81,7 +88,13 @@ document.querySelector('#pulse').addEventListener('click', changeColor);
 document.querySelector('#rainbow').addEventListener('click', changeColor);
 document.querySelector('#rainbowFade').addEventListener('click', changeColor);
 
+var colorChanging = false;
+
 function changeColor() {
+  if (colorChanging) {
```

Comment:
Need to clear colorChanging in the event of a write failing. All the .then(onColorChanged) need to have a .catch(_____).